pandas/core/groupby/grouper.py

```diff
@@ -516,6 +516,10 @@ def __init__(
                     t = self.name or str(type(self.grouper))
                     raise ValueError(f"Grouper for '{t}' not 1-dimensional")
                 self.grouper = self.index.map(self.grouper)
+
+                if isinstance(self.grouper, MultiIndex):
```

Comment:
why do you think we should support a MultiIndex grouper at all?

Reply:
1. Case for MultiIndex is already supported on line 440.
2. This bug is caused by grouper being identified as lambda. There might be other cases (line 515) where grouper will be re-identified and MultiIndex might be passed. In this case line 440 will be obsolete.

Reply:
As for use cases - creating a multi-dimensional groupby, that is generally supported and listed in docs. [Please, refer to comment under issue](https://github.com/pandas-dev/pandas/issues/36158#issuecomment-696940968). Lambda, in particular, for cases when group columns are not included in a DataFrame and some custom logic is needed. Sure thing, columns with groups can be added to the original DataFrame, but lambda gives another approach which right now experiences a bug.

Reply:
Last but not the least, grouping in general works well even with a bug present, only two functions are affected:
- indices
- get_group()

Also some other outputs described in diffs under [issue](https://github.com/pandas-dev/pandas/issues/36158#issuecomment-696940968) are affected as grouper uses MultiIndex where array is supposed to be present by current design (before modification in PR).